test/lit/passes/code-folding-eh.wast

```diff
@@ -0,0 +1,291 @@
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (return
+  ;; CHECK-NEXT:    (call $foo-i32)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (i32.const 0)
+  ;; CHECK-NEXT: )
+  (func $try-call-optimize-terminating-tails-call-return (result i32)
+    (block $tryend
+      (block $catch
+        (try_table (catch_all $catch)
+          (drop (i32.const 1))
+          (drop (i32.const 1))
+          ;; Cannot be folded out of the try because it might throw.
+          (return (call $foo-i32))
+        )
+        (br $tryend)
+      )
+      (drop (i32.const 1))
+      (drop (i32.const 1))
+      (return (call $foo-i32))
+    )
+    (i32.const 0)
+  )
+
+  ;; CHECK:      (func $try-call-optimize-terminating-tails-return-call (type $0) (result i32)
+  ;; CHECK-NEXT:  (block $folding-inner0
+  ;; CHECK-NEXT:   (return
+  ;; CHECK-NEXT:    (block (result i32)
+  ;; CHECK-NEXT:     (block $tryend
+  ;; CHECK-NEXT:      (block $catch
+  ;; CHECK-NEXT:       (try_table (catch_all $catch)
+  ;; CHECK-NEXT:        (br $folding-inner0)
+  ;; CHECK-NEXT:       )
+  ;; CHECK-NEXT:       (br $tryend)
+  ;; CHECK-NEXT:      )
+  ;; CHECK-NEXT:      (br $folding-inner0)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:     (i32.const 0)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (return_call $foo-i32)
+  ;; CHECK-NEXT: )
+  (func $try-call-optimize-terminating-tails-return-call (result i32)
+    (block $tryend
+      (block $catch
+        (try_table (catch_all $catch)
+          (drop (i32.const 1))
+          (drop (i32.const 1))
+          (drop (i32.const 1))
+          (return_call $foo-i32)
+        )
+        (br $tryend)
+      )
+      (drop (i32.const 1))
+      (drop (i32.const 1))
+      (drop (i32.const 1))
+      (return_call $foo-i32)
+    )
+    (i32.const 0)
+  )
+
+  ;; CHECK:      (func $try-call-optimize-expression-tails-success (type $1)
+  ;; CHECK-NEXT:  (block $x
+  ;; CHECK-NEXT:   (block $tryend
+  ;; CHECK-NEXT:    (block $catch
+  ;; CHECK-NEXT:     (try_table (catch_all $catch)
+  ;; CHECK-NEXT:      (br $x)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:     (br $tryend)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (br $x)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (unreachable)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $try-call-optimize-expression-tails-success
+    (block $x
+      (block $tryend
+        (block $catch
+          (try_table (catch_all $catch)
+            (drop (i32.const 1))
```

Comment:
Can you add a one-line comment saying expressions that cannot throw can be taken out of the try/try_table scope, both in here and the `eh-legacy` tests?